neo4j/internal/bolt/connect_test.go

```diff
@@ -186,3 +189,159 @@ func TestPerformManifestNegotiationNoSupportedVersion(t *testing.T) {
 		t.Errorf("Expected invalid handshake % X, got % X", expectedInvalid, fake.w.Bytes())
 	}
 }
+
+// fakeRacingReader is a simple implementation of a racing.RacingReader.
+type fakeRacingReader struct {
+	r *bytes.Reader
+}
+
+func newFakeRacingReader(data []byte) *fakeRacingReader {
+	return &fakeRacingReader{r: bytes.NewReader(data)}
+}
+
+func (f *fakeRacingReader) Read(_ context.Context, b []byte) (int, error) {
+	return f.r.Read(b)
+}
+
+func (f *fakeRacingReader) ReadFull(_ context.Context, b []byte) (int, error) {
+	return io.ReadFull(f.r, b)
+}
+
+// errorRacingReader always returns an error when Read is called.
+type errorRacingReader struct{}
+
+func (e *errorRacingReader) Read(_ context.Context, _ []byte) (int, error) {
+	return 0, fmt.Errorf("read error")
+}
+
+func (e *errorRacingReader) ReadFull(_ context.Context, b []byte) (int, error) {
+	return 0, fmt.Errorf("readfull error")
+}
+
+// TestEncodeVarInt tests that encodeVarInt returns the expected byte slices.
+func TestEncodeVarInt(t *testing.T) {
+	tests := []struct {
+		value    uint64
+		expected []byte
+	}{
+		{0, []byte{0x00}},
+		{1, []byte{0x01}},
+		{127, []byte{0x7F}},
+		{128, []byte{0x80, 0x01}},
+		{300, []byte{0xAC, 0x02}},
+		{16384, []byte{0x80, 0x80, 0x01}},
```

Comment:
```suggestion
		{16384, []byte{0x80, 0x80, 0x01}},
		{^uint64(0), []byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x01}},
```
Always a good idea to test the edge cases where possible :)